platform/src/main/java/org/hillview/storage/ParquetFileWriter.java

```diff
@@ -0,0 +1,115 @@
+package org.hillview.storage;
+
+import org.apache.avro.generic.GenericData;
+import org.apache.hadoop.fs.Path;
+import org.apache.parquet.avro.AvroParquetWriter;
+import org.apache.parquet.avro.AvroSchemaConverter;
+import org.apache.parquet.hadoop.ParquetWriter;
+import org.apache.parquet.hadoop.metadata.CompressionCodecName;
+import org.apache.parquet.schema.LogicalTypeAnnotation;
+import org.apache.parquet.schema.MessageType;
+import org.apache.parquet.schema.PrimitiveType;
+import org.apache.parquet.schema.Types;
+import org.hillview.table.ColumnDescription;
+import org.hillview.table.Schema;
+import org.hillview.table.api.IColumn;
+import org.hillview.table.api.IRowIterator;
+import org.hillview.table.api.ITable;
+
+import java.io.IOException;
+import java.util.List;
+
+public class ParquetFileWriter implements ITableWriter {
+    private final String filepath;
+
+    public ParquetFileWriter(String filepath) {
+        this.filepath = filepath;
+    }
+
+    private static MessageType getMessageType(Schema schema) {
+        Types.MessageTypeBuilder builder = Types.buildMessage();
+
+        for (ColumnDescription description : schema.getColumnDescriptions()) {
+            switch (description.kind) {
```

Comment:
I think that we should use native parquet types when they exist. E.g. Dates. Otherwise loading this data into an external dB will make no sense. What does the orc writer do?

Reply:
The orc writer does try to use native types if exists. I'll try to do that as well.

Reply:
Added native logical type support. Also added a new test that checks reading back after write gives the same result